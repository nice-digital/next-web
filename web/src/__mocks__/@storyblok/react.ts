export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
