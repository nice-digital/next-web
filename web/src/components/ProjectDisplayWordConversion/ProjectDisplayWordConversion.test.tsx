import { render, screen, within } from "@testing-library/react";

import mockConvertedDocumentFeed from "./../../__mocks__/__data__/inDev/guidance/gid-dg10086/documents/final-scope-html-conversion/html.json";
import {
	ProjectDisplayWordConversion,
	type ProjectDisplayWordConversionProps,
} from "./ProjectDisplayWordConversion";

const props: ProjectDisplayWordConversionProps = {
	content: mockConvertedDocumentFeed.content,
	sections: mockConvertedDocumentFeed.sections,
	inPageNavLinks: [],
	pdfLink: "/guidance/GID-DG10086/documents/final-scope-pdf",
	currentChapter: "",
	currentUrl:
		"/guidance/topic-selection/gid-dg10086/documents/final-scope-html-conversion",
};

describe("ProjectDisplayWordConversion", () => {
	it("should match snapshot for all nav items", () => {
		const { container } = render(<ProjectDisplayWordConversion {...props} />);
		expect(container).toMatchSnapshot();
	});

	it("should render the sections nav with two chapter menu items", () => {
		render(<ProjectDisplayWordConversion {...props} />);
		const chapterNavList = screen.getByRole("navigation", { name: "chapters" });
		const { getAllByRole } = within(chapterNavList);
		const chapterNavListItems = getAllByRole("listitem");
		expect(chapterNavList).toBeInTheDocument();
		expect(chapterNavListItems.length).toEqual(props.sections.length);
	});

	it("should render the first chapter menu item as active with index as slug", () => {
		render(<ProjectDisplayWordConversion {...props} />);
		const chapterNavList = screen.getByRole("navigation", { name: "chapters" });
		const { getAllByRole } = within(chapterNavList);
		const chapterNavListItems = getAllByRole("listitem");
		const { getByRole } = within(chapterNavListItems[0]);
		const firstNavListItemLink = getByRole("link", { current: true });
		expect(firstNavListItemLink).toBeInTheDocument();
	});

	it("should render the correct chapter menu item as active with chapter name as slug", () => {
		const dupedProps = { ...props };
		dupedProps.currentChapter = "recommendations";
		dupedProps.currentUrl =
			"/guidance/topic-selection/gid-dg10086/documents/final-scope-html-conversion/chapter/recommendations";
		render(<ProjectDisplayWordConversion {...dupedProps} />);
		const chapterNavList = screen.getByRole("navigation", { name: "chapters" });
		const { getByRole } = within(chapterNavList);
		const recommendationsNavListItem = getByRole("link", {
			name: "Recommendations",
			current: true,
		});
		expect(recommendationsNavListItem).toBeInTheDocument();
	});

	it("should render the html content properly", () => {
		render(<ProjectDisplayWordConversion {...props} />);
		const htmlContentHeading = screen.getByText("Overview", { selector: "h2" });
		const htmlContentParagraph = screen.getByText(
			/^(This guideline covers care)/i,
			{ selector: "p" }
		);
		expect(htmlContentHeading).toBeInTheDocument();
		expect(htmlContentParagraph).toBeInTheDocument();
	});

	it("should have prev and next links", () => {
		render(<ProjectDisplayWordConversion {...props} />);
		const prevAndNextLinksNav = screen.getByRole("navigation", {
			name: "Previous and next pages",
		});
		expect(prevAndNextLinksNav).toBeInTheDocument();
	});

	it("should have only next link when in first chapter", () => {
		render(<ProjectDisplayWordConversion {...props} />);
		const prevAndNextLinksNav = screen.getByRole("navigation", {
			name: "Previous and next pages",
		});
		const { getAllByRole } = within(prevAndNextLinksNav);
		const prevAndNextLinks = getAllByRole("link");
		expect(prevAndNextLinks.length).toEqual(1);
		expect(prevAndNextLinks[0].textContent).toContain("Next page");
	});

	it("should have only previous link when in last chapter", () => {
		const dupedProps = { ...props };
		dupedProps.currentChapter = "recommendations";
		dupedProps.currentUrl =
			"/guidance/topic-selection/gid-dg10086/documents/final-scope-html-conversion/chapter/recommendations";
		render(<ProjectDisplayWordConversion {...dupedProps} />);
		const prevAndNextLinksNav = screen.getByRole("navigation", {
			name: "Previous and next pages",
		});
		const { getAllByRole } = within(prevAndNextLinksNav);
		const prevAndNextLinks = getAllByRole("link");
		expect(prevAndNextLinks.length).toEqual(1);
		expect(prevAndNextLinks[0].textContent).toContain("Previous page");
	});

	it("should render pdf download button for converted document", () => {
		render(<ProjectDisplayWordConversion {...props} />);
		const downloadPDFButton = screen.getByRole("link", {
			name: "Download (PDF)",
		});
		expect(downloadPDFButton).toBeInTheDocument();
	});
});
