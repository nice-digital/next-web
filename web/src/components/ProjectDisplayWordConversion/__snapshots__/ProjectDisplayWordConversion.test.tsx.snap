// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDisplayWordConversion should match snapshot for all nav items 1`] = `
<div>
  <div
    class="grid grid--loose"
  >
    <section
      data-component="grid-item"
      data-g="12 md:4 lg:3"
    >
      <a
        aria-label="Download (PDF)"
        class="btn btn--cta download"
        data-component="button--cta"
        href="/guidance/GID-DG10086/documents/final-scope-pdf"
        target="_blank"
      >
        Download (PDF)
      </a>
      <nav
        aria-label="chapters"
        class="stacked-nav"
        data-component="stacked-nav"
      >
        <ul
          class="stacked-nav__list"
        >
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="true"
              href="/guidance/topic-selection/gid-dg10086/documents/final-scope-html-conversion"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                Overview
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/guidance/topic-selection/gid-dg10086/documents/final-scope-html-conversion/chapter/recommendations"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                Recommendations
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </section>
    <section
      data-component="grid-item"
      data-g="12 md:8 lg:9"
    >
      <h2
        class="mt--0"
      >
        Test converted document
      </h2>
      <div
        class="chapterContent"
      >
        <div
          class="chapter"
          id="0-1-overview"
          title="Overview"
          xmlns="http://www.w3.org/1999/xhtml"
        >
          
  
          <h2
            class="title"
          >
            
    
            <a
              class="annotator-chapter"
              data-heading-type="chapter"
              id="overview"
              style="position:relative"
            />
            Overview
            <span
              class="annotator-adder"
            />
          </h2>
          
  
          <p>
            This guideline covers care for women of any age (including girls and young women under 18) who request an abortion. It aims to improve the organisation of services and make them easier for women to access. Detailed recommendations on conducting abortions at different gestational stages are also included, to ensure that women get the safest and most effective care possible. 
          </p>
          

        </div>
      </div>
      <div
        class="prev-next"
      >
        <nav
          aria-label="Previous and next pages"
        >
          <a
            class="prev-next__link"
            href="/guidance/topic-selection/gid-dg10086/documents/final-scope-html-conversion/chapter/recommendations"
          >
            <span
              class="prev-next__link-intro"
            >
              Next page
            </span>
            <span
              class="prev-next__link-text"
            >
              Recommendations
            </span>
          </a>
        </nav>
      </div>
    </section>
  </div>
</div>
`;
